target/linux/octeon/image/Makefile

```diff
@@ -47,6 +51,18 @@ define Device/ubnt_edgerouter
 endef
 TARGET_DEVICES += ubnt_edgerouter
 
+define Device/ubnt_edgerouter-4
+  DEVICE_VENDOR := Ubiquiti
+  DEVICE_MODEL := EdgeRouter 4
+  DEVICE_DTS_DIR := ../dts
```

Comment:
Does octeon already have DTS files upstream or is this mach-files only?
In the first case, we should use the same directory as kernel does.

Reply:
technically octeon have dts files available upstream but for much (much!) older targets.
https://github.com/torvalds/linux/tree/master/arch/mips/boot/dts/cavium-octeon
there is nothing related to cavium3 devices (related cn71xx.dtsi here).
 
is there any other way to make that happen beside making device tree files as a patches for linux kernel?
that would be huge step back in ability to maintain target devices.

Reply:
The idea would be to just use directory "files/arch/mips/boot/dts/cavium-octeon" then instead of "dts".

Everything in "files" or "files-5.4" will be just added to the local kernel tree. This is used on many targets, e.g. https://github.com/openwrt/openwrt/tree/master/target/linux/ipq40xx/files/arch/arm/boot/dts

Reply:
do i need to do something for `files` to be added to local kernel tree? basically i just tried moving files from `dts` dir to `files/arch/mips/boot/dts/cavium-octeon`, and no files from `files` directory is copied to local kernel tree.

Reply:
Normally, this just works out-of-the-box. I'm not sure to which directory of the various directories containing the kernel it is copied exactly then.

Note that you might need adjustments in image/Makefile for building, though. This would done with target-specific DTS_DIR then, not with device-specific DEVICE_DTS_DIR...

Reply:
@adschm seems like for it to work i need to add a `generic` subtarget. that is the only difference between octeon and other targets and adding it - make `files` being able to be copied. if i remove it - files are not copied. will it be okay to add `generic` subtarget ?

Reply:
@adschm i found that files from <target>/files directory is copied to kernel directory only once when you do a fresh kernel build. rebuilds does not carry new files over. this is a huge way back from separate `dts` directory.

Reply:
i juar migrated device tree files to `files/arch/mips/boot/dts/cavium-octeon` and rebased against it.

Reply:
> i found that files from target/files directory is copied to kernel directory only once when you do a fresh kernel build. rebuilds does not carry new files over. 

That's strange, because for others targets it typically just worked to pull a PR adding a new DTS file, select it in make menuconfig and build.